# MetaDEGalaxy
MetaDEGalaxy is a metagenomic analysis tool running on Galaxy platform. There is a number of scripts written in R and python to make up the Galaxy workflow to perform 16S rRNA analysis. The greengenes database is used as a reference database in the MetaDEGalaxy.


## References
Mike W. C. Thang, X. Chua, G. Price, Dominique Gorse, Matthew A Field, 2019. [Metadegalaxy](https://f1000research.com/articles/8-726): Galaxy Workflow For Differential Abundance Analysis Of 16S Metagenomic Data.

# Tool
* [PEAR](https://cme.h-its.org/exelixis/web/software/pear/) - Paired-End reAd mergeR
* [VSEARCH](https://github.com/torognes/vsearch) - It is an alternative to the USEARCH tool

# Script dependencies:
* Biopython (v 1.74)
* python >=3
* Bioconductor-phyloseq (v 1.22.3)
* r-base (v 3.4.1)
* r-getopt (v 1.20.0)
* ghostscript (v 9.18)
* bioconductor-deseq2 (v 1.18.1)
* r-data.table (v 1.10.0)


# Script
* uc2otutable	- a python script to transform UC file format to tabular format. Note: [UC file format](https://www.drive5.com/usearch/manual/opt_uc.html)
* pear_stats	- a python script to generate an overlapping statistic report using the log file generated by PEAR.
* reheader		- a python script to rename the read identifier by appending the sample name to end of the read identifier.
* phyloseq_richness - a R package [phyloseq](https://joey711.github.io/phyloseq/) is used to generate richness plot.
* phyloseq_abundance_taxonomy - a script using R package [phyloseq](https://joey711.github.io/phyloseq/) to generata abundance plot by taxonomy.
* phyloseq_abundance_factor   - a script R using package [phyloseq](https://joey711.github.io/phyloseq/) to generate abundance plot by factor.
* phyloseq_2_deseq2   - a script using R package [phyloseq](https://joey711.github.io/phyloseq/) and [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) to generate differential abundance count table.
* phyloseq_net  - a script using R package [phyloseq](https://joey711.github.io/phyloseq/) to generate network plot based on the sample distance matrix.
* symmetric_plot - using [ggplot2](https://cran.r-project.org/web/packages/ggplot2/index.html) to generate a symmetric plot 


# Ansible script
This ansible script contains all the tools, scripts and Galaxy wrappers for MetaDeGalaxy. It can be used to install on a Galaxy Instance ( GVL 4.1.0 / ubuntu 14.04). More informatin can be found [here](https://github.com/QFAB-Bioinformatics/jcu.microgvl.ansible.playbook).

# Data
The testing data for MetaDEGalaxy is obtained from [mothur](https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip) website. More information about the dataset can be found on the [mothur](https://mothur.org/wiki/miseq_sop/) MiSeq SOP Getting started section.

# Tutorial
* Galaxy workflow tutorial can be found [here](https://github.com/QFAB-Bioinformatics/jcu.microgvl.ansible.playbook/tree/master/tutorial)

# Additional python script
One of the python scripts are obtained from [USearch](https://www.drive5.com/python/). The script uc2otutable.py was modified to run in python3 environment.


